platform/cervantes/disable-usbms.sh

```diff
@@ -23,4 +23,3 @@ mount -o "${MOUNT_ARGS}" -t vfat "${PARTITION}" /mnt/onboard
 PARTITION=${DISK}1p1
 
 [ -e "${PARTITION}" ] && mount -o "${MOUNT_ARGS}" -t vfat "${PARTITION}" /mnt/sd
-
```

Comment:
I'd rather keep the newlines at end of file.

Reply:
OK, but it's a shfmt issue (with a newer version: `3.7.0`), so how do you tell it to ignore those?

Out of curiosity, what's the rational?

Reply:
> Out of curiosity, what's the rational?

Better diffs, consistency, etc.

Reply:
I guess we can use `--ignore-blank-lines` in `base/utils/shellcheck.sh`
```diff
--- i/utils/shellcheck.sh
+++ w/utils/shellcheck.sh
@@ -40,7 +40,7 @@ shfmt() {
 }
 
 diff() {
-    local cmd=(command diff --unified)
+    local cmd=(command diff --ignore-blank-lines --unified)
     if [[ -n "${CLICOLOR_FORCE}" ]]; then
         cmd+=(--color=always)
     fi
```

Reply:
> OK, but it's a shfmt issue (with a newer version: `3.7.0`), so how do you tell it to ignore those?

I can't confirm; the behavior is the same in all versions I've tested (2.6.0, 3.6.0, 3.7.0, 3.8.0). It's a lack of newline that shows up in the diff, which is exactly what's desired.

Reply:
> It also removes leading or trailing empty lines, though, so if your file ends with two or more newlines, those will be replaced with a single newline.

So there must be a trailing new line, and only one.

Reply:
Yes, which is exactly the situation in those files.

Reply:
Interesting. ReaderFooter file didn't have a new line in the end when I started. Should I add one? Ref. https://github.com/koreader/koreader/pull/11999

Reply:
Nevermind, it probably did - I might have confused it along the way.

Reply:
It does; you'd be able to clearly see it on GH if it didn't. They add a red 🛇 or similar to indicate it.

But to be entirely correct, a single newline isn't trailing. A line is not a full and proper line without it: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206

Reply:
Those files have 2 trailing new lines.

And all version of shfmt complain.

It's the way the check was done previously, which would allow any number of newlines (including none):
```bash
▸ printf 'one newline\n' >t
▸ $shfmt -i 4 -ci -d t
▸ [[ "$(cat t)" = "$($shfmt -i 4 -ci t)" ]]; echo $?
0

▸ printf 'two newline\n\n' >t
▸ $shfmt -i 4 -ci -d t
--- t.orig
+++ t
@@ -1,2 +1,1 @@
 two newline
-
▸ [[ "$(cat t)" = "$($shfmt -i 4 -ci t)" ]]; echo $?
0

▸ printf 'five newline\n\n\n\n\n' >t
▸ $shfmt -i 4 -ci -d t
--- t.orig
+++ t
@@ -1,5 +1,1 @@
 five newline
-
-
-
-
▸ [[ "$(cat t)" = "$($shfmt -i 4 -ci t)" ]]; echo $?
0

▸ printf 'no newline' >t
▸ $shfmt -i 4 -ci -d t
--- t.orig
+++ t
▸ [[ "$(cat t)" = "$($shfmt -i 4 -ci t)" ]]; echo $?
0
```

Reply:
> Those files have 2 trailing new lines.
> 
> And all version of shfmt complain.

They don't and they don't.

Reply:
If you insist of your definition of line below, they end with 2 newlines characters, or one trailing empty line.

And yes, shfmt 3.4.3, 3.7.0, and 3.8.0 complain about it.

Reply:
Complain how?

```
# as is
$ shfmt_v3.8.0_linux_amd64 -i 4 enable-usbms.sh | diff enable-usbms.sh -
```

```
# no newline
$ shfmt_v3.8.0_linux_amd64 -i 4 enable-usbms.sh | diff enable-usbms.sh -
33c33
< sleep 1
\ No newline at end of file
---
> sleep 1
```

```
# two newline
$ shfmt_v3.8.0_linux_amd64 -i 4 enable-usbms.sh | diff enable-usbms.sh -
34d33
<
```

Reply:
It's `disable-usbms.sh`:
```bash
▸ tail -3 platform/cervantes/disable-usbms.sh

[ -e "${PARTITION}" ] && mount -o "${MOUNT_ARGS}" -t vfat "${PARTITION}" /mnt/sd

▸ for v in 3.4.3 3.7.0 3.8.0; do ./bin/shfmt_v${v}_linux_amd64 -i 4 -ci platform/cervantes/disable-usbms.sh | diff -u platform/cervantes/disable-usbms.sh -; done
--- platform/cervantes/disable-usbms.sh 2024-06-13 22:59:13.312754354 +0200
+++ -   2024-06-13 23:03:42.134574804 +0200
@@ -23,4 +23,3 @@
 PARTITION=${DISK}1p1

 [ -e "${PARTITION}" ] && mount -o "${MOUNT_ARGS}" -t vfat "${PARTITION}" /mnt/sd
-
--- platform/cervantes/disable-usbms.sh 2024-06-13 22:59:13.312754354 +0200
+++ -   2024-06-13 23:03:42.136661755 +0200
@@ -23,4 +23,3 @@
 PARTITION=${DISK}1p1

 [ -e "${PARTITION}" ] && mount -o "${MOUNT_ARGS}" -t vfat "${PARTITION}" /mnt/sd
-
--- platform/cervantes/disable-usbms.sh 2024-06-13 22:59:13.312754354 +0200
+++ -   2024-06-13 23:03:42.139547704 +0200
@@ -23,4 +23,3 @@
 PARTITION=${DISK}1p1

 [ -e "${PARTITION}" ] && mount -o "${MOUNT_ARGS}" -t vfat "${PARTITION}" /mnt/sd
-
```

Reply:
Ohhhkay, giant miscommunication. I thought you were removing the trailing newline but you were removing an additional improperly trailing newline.

But why aren't we seeing it in the output right now? As stated, every version behaves exactly the same in this regard.

Reply:
Because of the way the check was done with `[[ "$(cat t)" = "$($shfmt -i 4 -ci t)" ]]`, see comment above.

Reply:
> I thought you were removing the trailing newline but you were removing an additional improperly trailing newline.

This incidentally is exactly the reason it should be there, so as not to produce diffs that are distracting at best and misleading at worst. But clearly I should head off to bed. ;-)